pilot/pkg/config/kube/ior/route.go

```diff
@@ -531,60 +183,117 @@ func (r *route) createRoute(metadata config.Meta, gateway *networking.Gateway, o
 			TLS:            tlsConfig,
 			WildcardPolicy: wildcard,
 		},
-	}, metav1.CreateOptions{})
+	}
+}
+
+func (r *routeController) createRoute(
+	metadata config.Meta,
+	originalHost string,
+	tls *networking.ServerTLSSettings,
+	serviceNamespace, serviceName string,
+) (*v1.Route, error) {
+	IORLog.Debugf("creating route for hostname %s", originalHost)
+
+	nr, err := r.
+		routeClient.
+		RouteV1().
+		Routes(serviceNamespace).
+		Create(context.TODO(), buildRoute(metadata, originalHost, tls, serviceNamespace, serviceName), metav1.CreateOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating a route for the host %s from gateway: %s/%s",
+			originalHost,
+			metadata.Namespace,
+			metadata.Name)
+	}
+
+	IORLog.Infof("route %s/%s created for hostname %s from gateway %s/%s",
+		nr.ObjectMeta.Namespace, nr.ObjectMeta.Name,
+		nr.Spec.Host,
+		metadata.Namespace, metadata.Name)
+
+	return nr, nil
+}
+
+func (r *routeController) updateRoute(
+	metadata config.Meta,
+	originalHost string,
+	tls *networking.ServerTLSSettings,
+	serviceNamespace string, serviceName string,
+	route *v1.Route,
+) (*v1.Route, error) {
+	IORLog.Debugf("updating route for hostname %s", originalHost)
+
+	buildRoute(metadata, originalHost, tls, serviceNamespace, serviceName).DeepCopyInto(route.DeepCopy())
```

Comment:
Does this do anything? I don't think it does.

It builds a new route and copies it into a copy of `route`. But that copy isn't stored anywhere. In the update request that follows, the original route is used, making the update a no-op.